### ISSUE TITLE
Better candidate matching

### ISSUE THREAD

**Issue by abingham** [abingham/decktape.io#14]
Right now the candidate matching is pretty dumb, just levenshtein distance from the target url. This means that literally everything matches to some degree. We should figure out a better approach that takes into account what the user means. This will probably involve matching on domains or perhaps on path segments...something interesting.

**Comment by abingham** [comment]
Here's one [resource of algorithms](http://web.archive.org/web/20081224234350/http://www.dcs.shef.ac.uk/~sam/stringmetrics.html).

**Comment by abingham** [comment]
[Interesting, simple, levenshtein-based approach](http://stackoverflow.com/a/1472165/523939).

**Comment by abingham** [comment]
[stringscore](https://pypi.python.org/pypi/stringscore) seems pretty great.

**Comment by abingham** [comment]
Part of what we need to do for candidate matching is show only the most recent succesful conversion for a given URL. I think we can do that more quickly in mongo than in Python, and I think this snippet does the trick:

```
client = pymongo.MongoClient('localhost', 27017)
db = client.decktape_io
coll = db.file_ids
coll.aggregate([
    {'$match': {'metadata.status': 'complete'}},
    {'$sort': {'metadata.timestamp': -1}},
    {'$group': {'_id': '$metadata.url',
                'md': {'$first': '$metadata'}}}
])
```

Put this behind a new API in the result-db and test the heck out of it.

**Comment by abingham** [comment]
Fixed in fe9210ced95b76e7705d33ff7f191cb24df0582e
